feature_engine/base_transformers.py

```diff
@@ -154,17 +157,26 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
 
         return X
 
-    def get_feature_names_out(self) -> List:
+    @Substitution(
+        input_features=_input_features_docstring,
+    )
+    def get_feature_names_out(
+            self, input_features: Optional[Union[List, str]] = None
+    ) -> List:
         """Get output feature names for transformation.
 
+        Parameters
+        ----------
+        {input_features}
+
         Returns
         -------
         feature_names_out: list
             The feature names.
         """
         check_is_fitted(self)
-
-        return self.feature_names_in_
+        feature_names = check_input_features(input_features, self.variables_)
```

Comment:
we could probably just return the function without capturing it previously in a variable.

Have you read anything that says that one is better practice than the other?

Reply:
Do you mean changing the above code to `return check_input_features(input_features, self.variables_)`?

I never came across anything that suggested creating a number variable vs executing a function in a `return` statement. Also, I did some Googling; I didn't find anything.